src/ports/postgres/modules/deep_learning/madlib_keras.py_in

```diff
@@ -873,25 +901,42 @@ def internal_keras_eval_transition(state, dependent_var, independent_var,
         agg_loss = 0
         set_model_weights(segment_model, serialized_weights)
 
-    x_val = np_array_float32(independent_var, independent_var_shape)
-    y_val = np_array_int16(dependent_var, dependent_var_shape)
+    x_val = []
+    y_val = []
+    for counter, shape in enumerate(independent_var_shape):
+
+        x_val.append(np_array_float32(independent_var[counter], shape))
+    for counter, shape in enumerate(dependent_var_shape):
+
+        y_val.append(np_array_int16(dependent_var[counter], shape))
+
+    image_count = len(y_val[0])
+    agg_image_count += image_count
 
     with K.tf.device(device_name):
         res = segment_model.evaluate(x_val, y_val)
 
     # if metric is None, model.evaluate will only return loss as a scalar
     # Otherwise, it will return a list which has loss and metric
-    if type(res) is list:
-        loss, metric = res
+    if multi_output:
+        loss = res[0]
+        agg_loss += (image_count * loss)
+        for counter in range(output_count):
+            # For multi output cases, res has the following structure
+            # print(model.metrics_names)
+            # ['loss', 'dense_4_loss', 'dense_5_loss', 'dense_4_acc', 'dense_5_acc']
+            aux_losses[counter] = aux_losses[counter] + (image_count * res[counter+1])
+            aux_metrics[counter] = aux_metrics[counter] + (image_count * res[counter+1+len(dependent_var)])
```

Comment:
Wow, didn't realize eval was going to get this complicated!  Looks right though.